src/test/java/com/thebois/models/beings/ColonyTests.java

```diff
@@ -53,4 +65,179 @@ public void updateCallsUpdateOnAllBeings() {
+        // Arrange
+        final Colony colony = mockColony();
+
+        // Act
+        colony.add(new Log());
+        colony.add(new Log());
+
+        final int count = colony.numberOf(ItemType.LOG);
+
+        // Assert
+        assertThat(count).isEqualTo(2);
+    }
+
+    @Test
+    public void countIsCorrectValueWhenInventoryDoesNotHaveSpecifiedItemType() {
+        // Arrange
+        final Colony colony = mockColony();
+
+        // Act
+        colony.add(new Log());
+        colony.add(new Log());
+
+        final int count = colony.numberOf(ItemType.ROCK);
+
+        // Assert
+        assertThat(count).isEqualTo(0);
+    }
+
+    private Colony mockColony() {
+        final Collection<IBeing> pawns = new ArrayList<>(0);
+        return new Colony(pawns);
+    }
+
+    @Test
+    public void takeMultipleItemsFromColony() {
+        // Arrange
+        final Colony colony = mockColony();
+
+        // Act
+        colony.add(new Log());
+        colony.add(new Log());
+
+        final ArrayList<IItem> result = colony.takeAmount(ItemType.LOG, 2);
+
+        // Assert
+        assertThat(result.size()).isEqualTo(2);
+        assertThat(result.get(0).getType()).isEqualTo(ItemType.LOG);
+        assertThat(result.get(1).getType()).isEqualTo(ItemType.LOG);
+    }
+
+    @Test
+    public void canNotTakeMultipleItemsFromColony() {
+        // Arrange
+        final Colony colony = mockColony();
+
+        // Act
+        colony.add(new Log());
+        colony.add(new Log());
+
+        final Exception exception = assertThrows(IllegalArgumentException.class, () -> {
+            colony.takeAmount(ItemType.ROCK, 2);
+        });
+
+        // Assert
+        assertThat(exception.getMessage()).isEqualTo(
+            "Not enough of the specified ItemType in the inventory");
+    }
+
+    @Test
+    public void colonyHasItem() {
+        // Arrange
+        final Colony colony = mockColony();
+
+        // Act
+        colony.add(new Log());
+        final boolean result = colony.hasItem(ItemType.LOG);
+
+        assertThat(result).isTrue();
+    }
+
+    @Test
+    public void colonyDoesNotHaveItem() {
+        // Arrange
+        final Colony colony = mockColony();
+
+        // Act
+        final boolean result = colony.hasItem(ItemType.LOG);
+
+        assertThat(result).isFalse();
+    }
+
+    @Test
+    public void colonyHasMultipleOfItem() {
+        // Arrange
+        final Colony colony = mockColony();
+
+        // Act
+        colony.add(new Log());
+        colony.add(new Log());
+
+        final boolean result = colony.hasItem(ItemType.LOG, 2);
+
+        // Assert
+        assertThat(result).isTrue();
+    }
+
+    @Test
+    public void colonyDoesNotHaveMultipleOfItem() {
+        // Arrange
+        final Colony colony = mockColony();
+
+        // Act
+        final boolean result = colony.hasItem(ItemType.LOG, 2);
+
+        // Assert
+        assertThat(result).isFalse();
+    }
```

Comment:
question

Did you add a test case checking for the error we chatted about? Is this it or is it one of the other tests?

If I understood the error correctly, you were missing a test case for when there are items of the requested type in the inventory, but not enough. So exactly like this test here, but there is 1 LOG in the inventory.